New config entry + functionality

tarnishedmoth opened the pull request:
This patch adds a new config entry called **LockDodgeRolls**

It works by checking if LockDodgeRolls is true during the UpdateRotation switch AnimationState, and calling FaceLookDirection instead of doing nothing.

In game, the player model rotates with the camera even during a dodge-roll animation. This enhances PvP gameplay further by ensuring that other players can always see where you are aiming. This is necessary because the new crossbow items can be drawn long before firing, and the projectile always fires at your crosshair regardless of the player model's facing direction. In gameplay, this resulted in being able to 360 noscope ghostshot at the end of a dodge-roll.

tarnishedmoth committed:
added config entry for Lock Dodge Rolls
added if statement in FrozenAction animation switch-state

tarnishedmoth committed:
finalize

this works

tarnishedmoth commented:
Issue: Player can autorun with Q and free look still

gjroelofs commented:
> Issue: Player can autorun with Q and free look still

Is this something you are picking up or are we merging as is?

tarnishedmoth commented:
> > Issue: Player can autorun with Q and free look still
> 
> Is this something you are picking up or are we merging as is?

I'm going to try to fix it, and maybe rename the ConfigEntry to suit then.

tarnishedmoth committed:
Added config for ranged aim height
Changed config name for LockDodgeRolls to EnforceBodyRotation
Moved some config entries to Combat category
Removed !_meleeAimFix check in AttackStartPre, so that ranged attacks are always height-compensated by _configRangedAimHeightOffset in first-person mode.
Some tidying

tarnishedmoth committed:
Merge branch 'patch'

# Conflicts:
#	Client/FirstPersonValheimClientMod/FirstPersonValheimClientMod.cs
